back/engines/commercial/idea_custom_fields/app/controllers/idea_custom_fields/web_api/v1/admin/idea_custom_fields_controller.rb

```diff
@@ -113,6 +114,16 @@ def raise_error_if_stale_form_data
       raise UpdateAllFailedError, { form: [{ error: 'stale_data' }] }
     end
 
+    def raise_error_if_no_end_page_for_survey
+      is_survey = @custom_form.participation_context.pmethod.class.method_str == 'native_survey'
+      return unless is_survey
+
+      last_field = update_all_params.fetch(:custom_fields).last
+      return if last_field['key'] == 'survey_end' && last_field['input_type'] == 'page'
+
+      raise UpdateAllFailedError, { form: [{ error: 'no_end_page' }] }
+    end
+
```

Comment:
Probably not for now, but I guess if we can never destroy the last page then we should probably never be able to destroy the first page too? Good to be catching this at least. Assume the front-end does not permit it either